### PR TITLE
JENA-1643: Use transactional dataset in clear().

### PR DESCRIPTION
**Pull request by afs**
The fix is the one line change in method `clear()`.

### CODE REVIEW

File: jena-tdb/src/main/java/org/apache/jena/tdb/store/GraphTDB.java [hunk 1]
```diff
@@ -171,7 +171,10 @@ protected final int graphBaseSize() {
 
     @Override
     public void clear() {
-        getDatasetGraphTDB().deleteAny(getGraphName(), Node.ANY, Node.ANY, Node.ANY) ;
+        // Logically, this is "super.clear()" except the default implementation
+        // is a loop that materializes nodes. We want to call the dataset directly 
+        // so that nodes don't get materialized, just deleted from indexes.
```
Comment (ajs6f) [inline jena-tdb/src/main/java/org/apache/jena/tdb/store/GraphTDB.java]: @afs Just for my education, what do you mean by "materialized" here?

Comment (rvesse) [reply jena-tdb/src/main/java/org/apache/jena/tdb/store/GraphTDB.java]: Loaded from the node table into nodes in the cache I think?

Comment (afs) [reply jena-tdb/src/main/java/org/apache/jena/tdb/store/GraphTDB.java]: Yes.

**Comment by afs** [comment]
And review approvals?

**Pull request merged**